src/cuda/CMakeLists.txt

```diff
@@ -43,57 +42,30 @@ set(FINUFFT_CUDA_FLAGS
     >
     >)
 
-add_library(cufinufft_common_objects OBJECT ${PRECISION_INDEPENDENT_SRC})
-target_include_directories(cufinufft_common_objects
-                           PUBLIC ${CUFINUFFT_INCLUDE_DIRS})
-set_target_properties(
-  cufinufft_common_objects
-  PROPERTIES POSITION_INDEPENDENT_CODE ${FINUFFT_POSITION_INDEPENDENT_CODE}
-             CUDA_ARCHITECTURES "${FINUFFT_CUDA_ARCHITECTURES}"
-             CUDA_SEPARABLE_COMPILATION ON
-             CUDA_STANDARD 17
-             CUDA_STANDARD_REQUIRED ON)
-target_compile_features(cufinufft_common_objects PRIVATE cxx_std_17)
-target_compile_options(cufinufft_common_objects PRIVATE ${FINUFFT_CUDA_FLAGS})
-
-add_library(cufinufft_objects OBJECT ${PRECISION_DEPENDENT_SRC})
-target_include_directories(cufinufft_objects PUBLIC ${CUFINUFFT_INCLUDE_DIRS})
-set_target_properties(
-  cufinufft_objects
-  PROPERTIES POSITION_INDEPENDENT_CODE ${FINUFFT_POSITION_INDEPENDENT_CODE}
-             CUDA_ARCHITECTURES "${FINUFFT_CUDA_ARCHITECTURES}"
-             CUDA_SEPARABLE_COMPILATION ON
-             CUDA_STANDARD 17
-             CUDA_STANDARD_REQUIRED ON)
-target_compile_features(cufinufft_objects PRIVATE cxx_std_17)
-target_compile_options(cufinufft_objects PRIVATE ${FINUFFT_CUDA_FLAGS})
-
 if(FINUFFT_SHARED_LINKING)
-  add_library(cufinufft SHARED $<TARGET_OBJECTS:cufinufft_common_objects>
-                               $<TARGET_OBJECTS:cufinufft_objects>)
+  add_library(cufinufft SHARED ${PRECISION_INDEPENDENT_SRC} ${PRECISION_DEPENDENT_SRC})
```

Comment:
The reason for the logic before was to avoid compiling twice the sources! 
We know this could be done but then the compiler has to create the same objects twice :)